dashboard/generate_dashboard_data.py

```diff
@@ -0,0 +1,251 @@
+"""Manages dashboard data.
+Pulls event data from facebook, saves data as csvs in s3,
+loads data into the db, and generates data views."""
+import csv
+import io
+import json
+import hashlib
+import os
+import time
+
+import argparse
+from boto.s3.connection import S3Connection
+from boto.s3.key import Key
+import requests
+import psycopg2
+
+from airtable.airtable import get_all_records
+
+S3_BUCKET = "dxe-backup"
+S3_BACKUP_DIR = "dashboard"
+S3_ACCESS_KEY = os.environ["AIRTABLE_BACKUP_AWS_ACCESS_KEY_ID"]
+S3_SECRET_KEY = os.environ["AIRTABLE_BACKUP_AWS_SECRET_ACCESS_KEY"]
+
+
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in xrange(0, len(l), n):
+        yield l[i:i+n]
+
+
+def links_to_ids(links):
+    """Uses the Facebook Graph API to get group/page's ids from their links"""
+    # This uses's the batch request feature
+    # (https://developers.facebook.com/docs/graph-api/making-multiple-requests)
+    # to make many logical requests within one http request, saving the overhead
+    # cost
+
+    ids = []
+    for links_subset in chunks(links, 50):  # batch limit is 50
+        reqs = ['{{"method":"GET", "relative_url":"{}"}}'.format(link) for link in links_subset]
+        batch_req = '[' + ",".join(reqs) + ']'
+
+        r = requests.post(
+            "https://graph.facebook.com/v2.5",
+            data={"batch": batch_req},
+            headers={"Authorization": "Bearer {}".format(os.environ["FACEBOOK_APP_ACCESS_TOKEN"])}
+        )
+        data = r.json()
+        for i, res in enumerate(data):
+            community = json.loads(res['body'])['id']
+            if community.startswith("http"):
+                # If the id here is actually a link, then that means it was a fb group. If it's an
+                # id, then it was a fb page. The response format is different for each.
+                community = json.loads(res['body'])['og_object']['id']
+            ids.append(community)
+    return ids
+
+
+def get_facebook_chapter_links():
+    """Grab the facebook links for each local chapter from airtable."""
+    raw_data = get_all_records("Chapters", "Main View")
+    chapter_links = []
+    for row in raw_data:
+        fields = row["fields"]
+        if "Facebook" in fields:
+            chapter_links.append(fields["Facebook"])
+    return list(set(chapter_links))  # some differnent chapters use the same facebook page, remove dups
+
+
+def get_events(_id):
```

Comment:
add a link to the api somewhere? https://developers.facebook.com/docs/graph-api/reference/v2.5/event